api/pkg/transformer/feast/transformer.go

```diff
@@ -233,6 +224,51 @@ func (t *Transformer) getFeastFeature(ctx context.Context, request []byte, confi
 	}, nil
 }
 
+func buildEntitiesRequest(request []byte, configEntities []*transformer.Entity) ([]feast.Row, error) {
+	allEntityRows := [][]feast.Row{}
+	for _, entity := range configEntities {
+		vals, err := getValuesFromJSONPayload(request, entity)
+		if err != nil {
+			return nil, fmt.Errorf("unable to extract entity %s: %v", entity.Name, err)
+		}
+
+		rows := []feast.Row{}
+		for _, val := range vals {
+			rows = append(rows, feast.Row{
+				entity.Name: val,
+			})
+		}
+		allEntityRows = append(allEntityRows, rows)
+	}
+
+	if len(allEntityRows) == 0 {
+		return nil, fmt.Errorf("no entity extracted")
+	}
+
+	entities := allEntityRows[0]
+
+	entityIndex := 1
+	for entityIndex < len(configEntities) {
+		rows := allEntityRows[entityIndex]
+
+		newEntities := []feast.Row{}
+		for _, entity := range entities {
+			for _, row := range rows {
+				newFeastRow := feast.Row{}
+				for key, value := range entity {
+					newFeastRow[key] = value
+				}
+				newFeastRow[configEntities[entityIndex].Name] = row[configEntities[entityIndex].Name]
```

Comment:
it is better to extract `configEntities[entityIndex].Name` into a variable

Reply:
Can you elaborate? Are you saying we need validation as well?

Reply:
something like this
```
entityName = configEntities[entityIndex].Name
newFeastRow[entityName] = row[entityName]

```